src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/DatasourceRunner.java

```diff
@@ -141,12 +145,37 @@ protected void updateDatasource(final ScheduledJobParameter jobParameter, final
 
         try {
             datasourceUpdateService.deleteUnusedIndices(datasource);
-            datasourceUpdateService.updateOrCreateGeoIpData(datasource, renewLock);
+            if (DatasourceTask.DELETE_UNUSED_INDICES.equals(datasource.getTask()) == false) {
+                datasourceUpdateService.updateOrCreateGeoIpData(datasource, renewLock);
+            }
             datasourceUpdateService.deleteUnusedIndices(datasource);
         } catch (Exception e) {
             log.error("Failed to update datasource for {}", datasource.getName(), e);
             datasource.getUpdateStats().setLastFailedAt(Instant.now());
             datasourceFacade.updateDatasource(datasource);
+        } finally {
+            postProcessing(datasource);
+        }
+    }
+
+    private void postProcessing(final Datasource datasource) {
+        if (datasource.isExpired()) {
```

Comment:
If datasource is expired already, we don't want to schedule another task to delete expired datasource. Without this check, next call will trigger another task after 10 seconds and it will repeat again and again.